### PR TITLE
test(btc_parser): tx parser table test and edge cases

### PR DESCRIPTION
**Pull request by vuvoth**
## Description

Closes: #135
---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
    
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included doc comments for public functions
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary

## Summary by Sourcery

Add modular (de)serialization routines, taproot support, and reader-validation enforcement; refactor tx parsing logic and tests into a table-driven format, update documentation, and tighten build doc generation.

New Features:
- Add decode and encode functions for input and output to modularize (de)serialization
- Introduce new_witness constructor for transaction input witnesses
- Enforce reader exhaustion after transaction deserialization with ETxReaderNotEnd error
- Add taproot support with is_taproot and extract_taproot in the output module

Enhancements:
- Refactor tx.deserialize to use input::decode/encode and output::decode/encode and streamline raw transaction reconstruction
- Unify and parameterize transaction parsing tests into a table-driven approach
- Simplify documentation generation by adjusting Makefile-common.mk

Build:
- Adjust Makefile-common.mk to correct documentation copy path

Documentation:
- Document new decode/encode functions for input and output, new_witness for tx, and taproot-related functions in input.md, output.md, and tx.md

Tests:
- Add table-driven parser tests covering segwit, taproot, legacy transactions and expected_failure edge cases
- Add taproot detection tests in output module

### CODE REVIEW

File: packages/btc_parser/tests/tx_tests.move [hunk 1]
```diff
@@ -71,9 +236,27 @@ fun taproot_tx_parse_happy_cases() {
...
...000000ffffffff4d04ffff001d0104455468652054696d65732030332f4a616e2f32303039204368616e63656c6c6f72206f6e206272696e6b206f66207365636f6e64206261696c6f757420666f722062616e6b73ffffffff0100f2052a01000000434104678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5fac00000000010000000001010000000000000000000000000000000000000000000000000000000000000000ffffffff560338900d194d696e656420627920416e74506f6f6c20b9002000e1ba5728fabe6d6d8d05d057bb59e75b0a70582f882a227d93deb179034eb69fe356f7d9c53eda8a1000000000000000000024ff27b7020000000000ffffffff06220200000000000017a91442402a28dd61f2718a4b27ae72a4791d5bbdade7878c347c130000000017a9145249bdf2c131d43995cff42e8feee293f79297a8870000000000000000266a24aa21a9ed19d0d20ea5f2a543f9592bacde73a0955add5f612ebd77ef574f4cde80c52ed600000000000000002f6a2d434f524501a37cf4faa0758b26dca666f3e36d42fa15cc01064e3ecda72cb7961caa4b541b1e322bcfe0b5a0300000000000000000146a12455853415401000d130f0e0e0b041f12001300000000000000002b6a2952534b424c4f434b3ae54b94c2be8a749f45f6c3b2e1d30b5763f17e08c617ad91dd60d91000706ca501200000000000000000000000000000000000000000000000000000000000000000000000001111"; // add 1111
```
[the remark this answers is not in the document]
Comment (vuvoth) [reply packages/btc_parser/tests/tx_tests.move:249]: We have comments about what we need to change. So this is not necessary.
